### ISSUE TITLE
Change node setting JSON value uses one shared object

### ISSUE THREAD

**Issue by alexmojaki** [node-red/node-red-nodes#232]
I have a flow including a `change` node which sets some property of `msg` to a JSON object. Since the object is only parsed once and never copied, all messages get the same shared object, which leads to infuriatingly confusing bugs. Here's a flow demonstrating the problem:

```
[{"id":"e1591705.a4bf28","type":"inject","z":"cada6667.200e7","name":"","topic":"","payload":"","payloadType":"date","repeat":"","crontab":"","once":false,"x":193.5,"y":258,"wires":[["31062ec4.6425ea"]]},{"id":"253902e4.cedb16","type":"function","z":"cada6667.200e7","name":"","func":"msg.payload.a += 1;\nreturn msg;","outputs":1,"noerr":0,"x":587.5,"y":265,"wires":[["bbbd199d.06ab48"]]},{"id":"bbbd199d.06ab48","type":"debug","z":"cada6667.200e7","name":"","active":true,"console":"false","complete":"false","x":770.5,"y":259,"wires":[]},{"id":"31062ec4.6425ea","type":"change","z":"cada6667.200e7","name":"","rules":[{"t":"set","p":"payload","pt":"msg","to":"{\"a\": 1}","tot":"json"}],"action":"","property":"","from":"","to":"","reg":false,"x":399.5,"y":256,"wires":[["253902e4.cedb16"]]}]
```

**Comment by knolleary** [comment]
@alexmojaki good spot, thanks. 

fyi, all of the core nodes are maintained in the main repo - node-red/node-red. This repo is for the extra nodes we publish separately to npm.

**Comment by dceejay** [comment]
closed by fix in core node-red-project 0dd2c7fe243c9386414c7e2664f843259a2935c3
